### PR TITLE
minor fixes for Coverity reports

### PR DESCRIPTION
**Pull request by JacobBarthelmeh**
minor fixes for Coverity reports

### CODE REVIEW

**Comment by JacobBarthelmeh** [comment]
Retest this please Jenkins. Possible network timeout: PRB-single-flag.txt_95

```
SSL curve name is SECP256R1

Unexpected server result

removing ready file: /tmp/workspace/PRB-generic-config-parser/wolfssl/tmp.l0SKcXeQK1/wolf_ocsp_s2_readyF1162286

removing ready file: /tmp/workspace/PRB-generic-config-parser/wolfssl/tmp.l0SKcXeQK1/wolf_ocsp_s2_readyF2162286

removing ready file: /tmp/workspace/PRB-generic-config-parser/wolfssl/tmp.l0SKcXeQK1/wolf_ocsp_s2_readyF3162286

removing ready file: /tmp/workspace/PRB-generic-config-parser/wolfssl/tmp.l0SKcXeQK1/wolf_ocsp_s2_readyF4162286

removing ready file: /tmp/workspace/PRB-generic-config-parser/wolfssl/tmp.l0SKcXeQK1/wolf_ocsp_s2_readyF5162286

retrying...

FAIL scripts/ocsp-stapling_tls13multi.test (exit status: 143)

<<<

Error: Test runtime=4603sec. Make Check RESULT = 2

Exiting with status: 1

Script ran for 5167 seconds.

script returned exit code 1
```

File: src/x509.c [hunk 1]
```diff
@@ -14111,7 +14111,9 @@ int wolfSSL_X509_check_ip_asc(WOLFSSL_X509 *x, const char *ipasc,
     }
 
 #ifdef WOLFSSL_SMALL_STACK
-    XFREE(dCert, x->heap, DYNAMIC_TYPE_DCERT);
+    if (dCert != NULL) {
+        XFREE(dCert, x->heap, DYNAMIC_TYPE_DCERT);
```
[the remark this answers is not in the document]
Comment (JacobBarthelmeh) [reply src/x509.c:14115]: thanks, I should have checked x for null here not dCert

File: wolfcrypt/src/aes.c [hunk 2]
```diff
@@ -10531,8 +10531,12 @@ int wc_Gmac(const byte* key, word32 keySz, byte* iv, word32 ivSz,
 #else
     ret = wc_AesInit(aes, NULL, INVALID_DEVID);
 #endif
-    if (ret != 0)
+    if (ret != 0) {
+    #ifdef WOLFSSL_SMALL_STACK
+        wc_AesDelete(aes, NULL);
+    #endif
```
[the remark this answers is not in the document]
Comment (JacobBarthelmeh) [reply wolfcrypt/src/aes.c:10537]: likely a false positive then, I missed that the return would always be null if ret is set to nonzero

File: wolfcrypt/test/test.c [hunk 3]
```diff
@@ -35099,6 +35100,16 @@ WOLFSSL_TEST_SUBROUTINE wc_test_ret_t curve25519_test(void)
...
 
+    /* wc_FreeRng is always called on exit. Therefore wc_InitRng should be
+     * called before any exit goto's */
+#ifndef HAVE_FIPS
+    ret = wc_InitRng_ex(&rng, HEAP_HINT, devId);
+#else
+    ret = wc_InitRng(&rng);
+#endif
+    if (ret != 0)
+        ERROR_OUT(WC_TEST_RET_ENC_EC(ret), cleanup);
```
Comment (douzzer) [inline wolfcrypt/test/test.c:35111]: whoops, need to actually `return WC_TEST_RET_ENC_EC(ret)` here, since `cleanup:` will `wc_FreeRng()` unconditionally.

**Review by douzzer — CHANGES_REQUESTED** [review]

**Review by douzzer — APPROVED** [review]

**Pull request merged**